class.jetpack.php

```diff
@@ -5318,16 +5318,23 @@ function verify_xml_rpc_signature() {
 		}
 
 		// Let's see if this is onboarding. In such case, use user token type and the provided user id.
-		if ( isset( $this->HTTP_RAW_POST_DATA ) ) {
-			$jpo = json_decode( $this->HTTP_RAW_POST_DATA );
+		if ( isset( $this->HTTP_RAW_POST_DATA ) || ! empty( $_GET['onboarding'] ) ) {
+			if ( ! empty( $_GET['onboarding'] ) ) {
+				$jpo_token = ! empty( $_GET['onboarding']['token'] ) ? $_GET['onboarding']['token'] : null;
+				$jpo_user = ! empty( $_GET['onboarding']['jpUser'] ) ? $_GET['onboarding']['jpUser'] : null;
+			} else {
+				$jpo = json_decode( $this->HTTP_RAW_POST_DATA );
+				$jpo_token = ! empty( $jpo->onboarding->token ) ? $jpo->onboarding->token : null;
+				$jpo_user = ! empty( $jpo->onboarding->jpUser ) ? $jpo->onboarding->jpUser : null;
+			}
```

Comment:
Can we de-dupe some virtually equivalent logic if in the `_GET` case we simply cast `_GET['onboarding']` to an object (and use the same logic as in the `POST` case afterwards)? 

Also, maybe one variable could be enough? E.g. if we set `$jpo` to what is now `$jpo->onboarding`, we could use `$jpo->token` and `$jpo->jpUser` below. (Modulo a bunch of `isset` checks that we might have to modify.)

Reply:
Good call. Fixed in https://github.com/Automattic/jetpack/pull/8461/commits/ae1c3308ba2c7ed3cc97b64e4659378b671ec162

I don't like converting the `$_GET` to an object, so I went the other way around.